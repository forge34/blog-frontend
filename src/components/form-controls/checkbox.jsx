import React from "react";
import propTypes from "prop-types";

function Checkbox({ name, label }) {
  return (
    <div className="checkbox">
      <label htmlFor={name}>{label}</label>
      <input type="checkbox" name={name}></input>
    </div>
  );
}

Checkbox.propTypes = {
  name: propTypes.string,
  label: propTypes.string,
};

export default Checkbox;
