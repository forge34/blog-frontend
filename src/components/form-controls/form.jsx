import React from "react";
import PropTypes from "prop-types";

function Form({ handleSubmit = () => {}, children }) {
  return (
    <form
      onSubmit={(e) => {
        e.preventDefault();
        handleSubmit(e);
      }}
    >
      {children}
    </form>
  );
}

Form.propTypes = {
  children: PropTypes.element,
  handleSubmit: PropTypes.func,
};

export { Form };
