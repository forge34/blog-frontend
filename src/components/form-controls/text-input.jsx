import propTypes from "prop-types";
import { useState } from "react";

const TextInput = ({ name, label, handleChange = () => {} }) => {
  const [value, setValue] = useState("");

  return (
    <div>
      <label>
        {label}
        <input
          name={name}
          value={value}
          onChange={(e) => {
            handleChange(e);
            setValue(e.target.value);
          }}
        ></input>
      </label>
    </div>
  );
};

TextInput.propTypes = {
  name: propTypes.string,
  label: propTypes.string,
  handleChange: propTypes.func,
};

export { TextInput };
