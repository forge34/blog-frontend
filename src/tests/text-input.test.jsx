import { render, screen } from "@testing-library/react";
import { TextInput } from "../components/form-controls/text-input";
import userEvent from "@testing-library/user-event";

describe("", () => {
  it("Element is in the document", () => {
    render(<TextInput name="username" label="username"></TextInput>);

    expect(screen.getByLabelText("username")).toBeInTheDocument();
  });

  it("Element matches snap shot", () => {
    const { input } = render(
      <TextInput name="username" label="username"></TextInput>,
    );

    expect(input).toMatchSnapshot();
  });

  it("Element changes text", async () => {
    render(<TextInput name="username" label="username"></TextInput>);
    const field = screen.getByLabelText("username");

    const user = userEvent.setup();

    await user.type(field, "name");

    expect(field).toHaveValue("name");
  });
});
