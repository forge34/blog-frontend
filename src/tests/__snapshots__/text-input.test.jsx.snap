// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Element matches snap shot 1`] = `undefined`;
